womtool/src/main/scala/cromwell/pipeline/womtool/WomTool.scala

```diff
@@ -0,0 +1,78 @@
+package cromwell.pipeline.womtool
+
+import java.nio.file.{ Files, Paths }
+
+import cats.data.NonEmptyList
+import com.typesafe.config.ConfigFactory
+import common.Checked
+import cromwell.core.path.{ DefaultPath, DefaultPathBuilder, Path }
+import cromwell.languages.LanguageFactory
+import cromwell.languages.util.ImportResolver.ImportResolver
+import languages.cwl.CwlV1_0LanguageFactory
+import languages.wdl.biscayne.WdlBiscayneLanguageFactory
+import languages.wdl.draft2.WdlDraft2LanguageFactory
+import languages.wdl.draft3.WdlDraft3LanguageFactory
+import wom.executable.WomBundle
+import womtool.WomtoolMain.{ BadUsageTermination, SuccessfulTermination, Termination, UnsuccessfulTermination }
+import womtool.inputs.Inputs
+
+import scala.util.Try
+
+class WomTool extends WomToolAPI {
+
+  def validate(
+    content: String,
+    importResolvers: List[ImportResolver]
+  ): Either[NonEmptyList[String], (WomBundle, LanguageFactory)] = {
+
+    val languageFactory: LanguageFactory =
+      List(
+        new WdlDraft3LanguageFactory(ConfigFactory.empty()),
+        new WdlBiscayneLanguageFactory(ConfigFactory.empty()),
+        new CwlV1_0LanguageFactory(ConfigFactory.empty())
+      ).find(_.looksParsable(content)).getOrElse(new WdlDraft2LanguageFactory(ConfigFactory.empty()))
+
+    val bundle: Checked[WomBundle] =
+      languageFactory.getWomBundle(content, None, "{}", importResolvers, List(languageFactory))
+
+    def getBundleAndFactory(womBundle: Checked[WomBundle]): Either[NonEmptyList[String], (WomBundle, LanguageFactory)] =
+      womBundle match {
+        case Right(w) => Right(w, languageFactory)
+        case Left(l)  => Left(l)
+      }
+    getBundleAndFactory(bundle)
+  }
+
+  def inputs(content: String): Either[NonEmptyList[String], String] = {
+
+    def createTmpFile(wdlFileContent: String): Try[java.nio.file.Path] =
+      Try(Files.createTempFile("tmp", ".wdl"))
+
+    def writeTmpFile(path: java.nio.file.Path): Try[java.nio.file.Path] =
+      Try(Files.write(path, content.getBytes()))
+
+    def deleteTmpFile(tempFile: java.nio.file.Path): Unit =
+      Files.delete(tempFile)
+
+    def getTryiedPath(tmpPath: java.nio.file.Path): Try[DefaultPath] =
+      DefaultPathBuilder.build(tmpPath.toString)
+
+    def getTermination(path: Path): Termination = Inputs.inputsJson(path, false)
+
+    def getEither(term: Termination): Either[NonEmptyList[String], String] = term match {
+      case SuccessfulTermination(x)   => new Right(x)
+      case UnsuccessfulTermination(x) => new Left(NonEmptyList(x, Nil))
+      case BadUsageTermination(x)     => new Left(NonEmptyList(x, Nil))
+    }
+
+    val tmpPath: Try[DefaultPath] = for {
+      created <- createTmpFile(content)
+      written <- writeTmpFile(created)
+      tPath <- getTryiedPath(written)
+    } yield (tPath)
+
+    val either = getEither(getTermination(tmpPath.get))
+    deleteTmpFile(Paths.get(tmpPath.get.pathAsString))
+    either
+  }
```

Comment:
File system usage is a straightforward way to have a bunch of performance, memory, error-prone and security problems, I suggest we implement this logic by ourself without involving any `IO` operations (`inputNodeWriter` and `womTypeToJson` same as in the [original womtool](https://github.com/broadinstitute/cromwell/blob/4657460739bbcf6245c2a6f7bfbb53c10ead11f3/womtool/src/main/scala/womtool/inputs/Inputs.scala#L29-L45)):

```suggestion
  def inputs(content: String): Either[NonEmptyList[String], String] = {
    val languageFactory = languageFactories.find(_.looksParsable(content)).getOrElse(defaultLanguageFactory)

    for {
      bundle   <- languageFactory.getWomBundle(content, None, "{}", importResolvers, List(languageFactory))
      callable <- bundle.toExecutableCallable
      wdl = WomGraphWithResolvedImports(callable.graph, bundle.resolvedImportRecords)
    } yield wdl.graph.externalInputNodes.toJson(inputNodeWriter).prettyPrint + System.lineSeparator
  }

  private def inputNodeWriter: JsonWriter[Set[ExternalGraphInputNode]] = set => {
    val valueMap: Seq[(String, JsValue)] = set.toList.collect {
      case RequiredGraphInputNode(_, womType, nameInInputSet, _) => nameInInputSet -> womTypeToJson(womType, None)
      case OptionalGraphInputNode(_, womOptionalType, nameInInputSet, _) =>
        nameInInputSet -> womTypeToJson(womOptionalType, None)
      case OptionalGraphInputNodeWithDefault(_, womType, default, nameInInputSet, _) =>
        nameInInputSet -> womTypeToJson(womType, Option(default))
    }

    valueMap.toMap.toJson
  }

  private def womTypeToJson(womType: WomType, default: Option[WomExpression]): JsValue = (womType, default) match {
    case (WomCompositeType(typeMap, _), _) =>
      JsObject(
        typeMap.map { case (name, wt) => name -> womTypeToJson(wt, None) }
      )
    case (_, Some(d))            => JsString(s"${womType.stableName} (optional, default = ${d.sourceString})")
    case (_: WomOptionalType, _) => JsString(s"${womType.stableName} (optional)")
    case (_, _)                  => JsString(s"${womType.stableName}")
  }
```

**NB** We could refactor it later, at this point I believe the main goal is to have some working solution.

Reply:
complete